src/ring_theory/ideals.lean

```diff
@@ -228,6 +228,46 @@ def map_mk (I J : ideal α) : ideal I.quotient :=
   smul_mem' := by rintro ⟨c⟩ _ ⟨x, hx, rfl⟩;
     exact ⟨c * x, J.mul_mem_left hx, rfl⟩ }
 
+/-- The preimage of an ideal J under the quotient map `R → R/I`. -/
+def comap_mk (I : ideal α) (j : ideal I.quotient) : ideal α :=
+{ carrier := mk I ⁻¹' j,
+  zero_mem' := j.zero_mem,
+  add_mem' := λ _ _ hx hy, j.add_mem hx hy,
+  smul_mem' := λ c _ hx, by simp [j.mul_mem_left hx] }
+
+@[simp] lemma map_mk_top {I : ideal α} : map_mk I ⊤ = ⊤ :=
+le_antisymm le_top (by rintro ⟨x⟩ hx; use ⟨x, hx, rfl⟩)
+
+lemma map_mk_comap_mk_left_inverse {I : ideal α} : left_inverse (map_mk I) (comap_mk I) :=
```

Comment:
`map_mk` and `comap_mk` form a galois insertion. A lot of the lemmas here. can be proved very easily by first proving that `map_mk` and `comap_mk` form a galois insertion. It might be a galois coinsertion actually, which we don't have, in which case ignore this comment, because we don't have them yet. You could always define them, this wouldn't be too hard.